### DISCUSSION TITLE
Return value hooking

### DISCUSSION THREAD

**Discussion by marielMifsud** [pamidur/aspect-injector#205] [Q&A]
Hi, I was wondering is there something like a proceed() in AspectJ where I can get the return value of a method which I am trying to hook?

Thanks :)

**Comment by pamidur** [comment]
Hi @marielMifsud , yes it is there. You need to use ```Advice[Kind.Around]``` and arguments ```Target``` and ```Arguments```
```c#
 [Advice(Kind.Around)]
        public object Handle(
            [Argument(Source.Target)] Func<object[], object> target,
            [Argument(Source.Arguments)] object[] args
            )
        {
                var returnValue = target(arguments);
        }

```
